### PR TITLE
Bug 922956 - Add unit test to prevent regressions

### PR DESCRIPTION
**Pull request by jaragunde**
The bug seems to be fixed, I'm adding a unit test to prevent regressions and close the bug for good.

### CODE REVIEW

**Comment by benmccann** [comment]
@erikrose @Mossop @jsantell @Gozala would one of you guys be able to take a look at this to see if bug 922956 is really fixed?

**Comment by benmccann** [comment]
ping. would really love it if someone could take a look at this one

**Comment by erikrose** [comment]
I think you got the wrong Erik; you probably meant one of the 2 CCd on the ticket.

**Comment by benmccann** [comment]
Sorry, yep, I meant to tag @erikvold

**Comment by jaragunde** [comment]
I've slightly modified the test to include a call to window.open with some parameters to have that case covered.

File: test/test-window-events.js [hunk 1]
```diff
@@ -60,4 +64,20 @@ exports["test browser events"] = function(assert, done) {
   let window = open();
 };
 
+// related to bug 922956
+// https://bugzilla.mozilla.org/show_bug.cgi?id=922956
+exports["test ready event after window.open"] = function (assert, done) {
+  setPref(OPEN_IN_NEW_WINDOW_PREF, 2);
+  setPref(DISABLE_POPUP_PREF, false);
```
Comment (Mossop) [inline test/test-window-events.js]: You need to reset these prefs at the end of the test or it may affect later tests.

Comment (Mossop) [inline test/test-window-events.js]: That may be what is causing the travis tests to fail, if not it looks like they warrant investigation.

**Comment by jaragunde** [comment]
@Mossop Thanks for the review; I've added your suggestions and submitted a new patch. Unfortunately it did not pass Travis either; this time there is a weird error with no list of failed tests.

I'll keep looking into it.

**Comment by Mossop** [comment]
Looks like some timeout is happening or something. The first failure, mozRequestAnimationFrame is due to a change in Firefox, I've fixed that in master so you might try rebasing and pushing again to see if that fixes the others too.

**Comment by benmccann** [comment]
@jaragunde can you rebase and push again? Thanks!

**Comment by benmccann** [comment]
@Mossop looks like this PR was updated if you want to take another look

File: test/tabs/test-firefox-tabs.js [hunk 2]
```diff
@@ -1234,6 +1234,33 @@ exports['test active tab properties defined on popup closed'] = function (assert
...
+
+  let firstRun = true;
+  tabs.on('ready', function onReady(tab) {
+    if (firstRun) {
+      assert.pass("tab ready callback after 1st window.open");
+      firstRun = false;
+      tab.close();
+    }
+    else {
+      assert.pass("tab ready callback after 2nd window.open");
```
Comment (benmccann) [inline test/tabs/test-firefox-tabs.js]: wouldn't the test pass even if neither of the `assert.pass` lines were ever called?

how does this line get called? won't the test exit on the first call to `assert.pass`? (i'm not sure exactly what testing framework is being used here, but `assert.pass` will end a test as passing in most frameworks i've seen)

File: test/tabs/test-firefox-tabs.js [hunk 3]
```diff
@@ -1234,6 +1234,33 @@ exports['test active tab properties defined on popup closed'] = function (assert
...
+      tabs.removeListener('ready', onReady);
+      tab.close(done);
```
Comment (jaragunde) [inline test/tabs/test-firefox-tabs.js]: @benmccann The test will fail if it does not end with a call to done(). That call happens in the line above, after tab is closed.

File: test/tabs/test-firefox-tabs.js (hunk 2 above)
Comment (jaragunde) [inline test/tabs/test-firefox-tabs.js]: @benmccann In this case, assert.pass() just logs a message.

**Comment by jaragunde** [comment]
Travis has failed again, but all the tests pass successfully on my local machine. Besides, the same set of tests are failing in another [unrelated pull request](https://travis-ci.org/mozilla/addon-sdk/builds/74027446), so I suspect the problem may not be related with this patch.

**Comment by Mossop** [comment]
Yeah I ran try builds against the full Firefox tree yesterday and they came up green so something is broken with travis right now. No need to hold this up any longer, sorry for the caution there.

**Pull request merged**